File: leetcode/0008-string-to-integer-atoi/solution.py
class Solution(object):
    def myAtoi(self, s):
        s=s.strip()
        p=''
        x=True
        if s=='':
            return 0
        elif s[0]=='+' and len(s)>=2:
            if s[1] in "1234567890":
                for i in s[1:]:
                    if i in "1234567890":
                        p=p+i
                    else:
                        break
        elif s[0]=='-':
            x=False
            for i in s[1:]:
                if i in "1234567890":
                    p=p+i
                else:
                    break
        elif s[0] in "1234567890":
            for i in s:
                if i in "1234567890":
                    p=p+i
                else:
                    break
        if p!='':
            if x==True:
                p=int(p)
            else:
                p=int(p)*-1
            if p<-2**31:
                return -2**31
            elif p>2**31 -1:
                return 2**31 -1
            else:
                return p
        else:
            return 0
        """
        :type s: str
        :rtype: int
        """
        

File: leetcode/0008-string-to-integer-atoi/solution.cpp
class Solution {
public:
    int helper(string s, int i,double sum,int sign){
        if(i>=s.size() || !isdigit(s[i])){
            return sum*sign;
        }
        sum=sum*10 + (int)s[i]-48;

        if(sign==1&&sum>INT_MAX) return INT_MAX;
        if(sign==-1&&-sum<INT_MIN) return INT_MIN;

        return helper(s,i+1,sum,sign);

    }
    int myAtoi(string s) {
        int i=0;
        double sum=0;
        int sign=1;
        while(i<s.size() && s[i]==' ') i++;
        if(i<s.size() && (s[i]=='-' || s[i]=='+')){
            if(s[i]=='-') sign=-1;
            i++;
        }
        sum=helper(s,i,sum,sign);
        return sum;
    }
};
